include/blas_helper.cuh

```diff
@@ -10,35 +10,14 @@
 namespace quda
 {
 
-  template <bool X_ = false, bool Y_ = false, bool Z_ = false, bool W_ = false, bool V_ = false> struct write {
+  template <bool X_ = false, bool Y_ = false, bool Z_ = false, bool W_ = false, bool V_ = false> struct stream {
```

Comment:
just wondering whether stream is the best name. I don't currently have a better suggestion.

Reply:
`mem_out`? Not saying that's better, but I agree we should avoid using "stream" for anything that isn't a CUDA/etc stream...

Reply:
It's for both input and output, so `out` in the type name isn't appropriate.  `memory_stream` perhaps?

Reply:
Sure, or `mem_io`. Depends on how pedantic we want to be about not using `stream`, even from a `grep` sense.

Reply:
how about calling it `memory_access` ? Would make something like `memory_access<1,1,0,0,0> read; memory_access<1,0,0,0,0> write` ? Guess if we would live in a FORTRAN world `INTENT` would be the proper name.

Reply:
Ok, `memory_access` it is.  Will plumb this in with next push.